CE-468 - Update Consul recommended hardware sizing (#1666)
hashicorp/web-unified-docs: content/consul/v1.21.x/content/docs/reference/architecture/server.mdx

| 8-16 core | 32-64 GB RAM | 200+ GB       | 7500+ IOPS  | 250+ MB/s       | Lower than 50ms     | Lower than 100ms    |

These requirements will vary based on the size of the Consul cluster, the number of client agents, and the workload characteristics (e.g. read-heavy vs write-heavy). While Consul may run on smaller hardware configurations, the above specifications are recommended to ensure optimal performance and reliability in production environments, and will help avoid having to upgrade hardware during a capacity outage.

The following table gives a recommended starting point for hardware specs for each major cloud infrastructure provider.

| Provider  | Instance/VM Types                     | Disk Volume Specs                          |
| --------- | ------------------------------------- | ------------------------------------------ |
| **AWS**   | `m5.2xlarge`, `m5.4xlarge`            | 200+GB `gp3`, 10000 IOPS, 250MB/s          |
| **Azure** | `Standard_D8s_v3`, `Standard_D16s_v3` | 2048GB\* `Premium SSD`, 7500 IOPS, 200MB/s |
| **GCP**   | `n2-standard-8`, `n2-standard-16`     | 1000GB\* `pd-ssd`, 30000 IOPS, 480MB/s     |

> ## RAFT Performance Tuning Defaults

Reviewer: ```suggestion
## Raft performance tuning defaults
```

[RAFT](/consul/docs/concept/consensus) is the consensus algorithm that Consul nodes use. The default server [performance parameters](/consul/docs/reference/agent/configuration-file/general#performance) are tuned to allow Consul to run reliably (but relatively slowly) on a server cluster of three [AWS t2.micro](https://aws.amazon.com/ec2/instance-types/) instances. These thresholds were determined empirically using a leader instance that was under sufficient read, write, and network load to cause it to permanently be at zero CPU credits, forcing it to the baseline performance mode for that instance type. Real-world workloads typically have more bursts of activity, so this is a conservative and pessimistic tuning strategy.

> This default was chosen based on feedback from users, many of whom wanted a low cost way to run small production or development clusters with low cost compute resources, at the expense of some performance in leader failure detection and leader election times.

Reviewer: ```suggestion
We chose these defaults so that you can run small production or development clusters with lower cost compute resources, at the expense of some performance in leader failure detection and leader election times.
```

> The default RAFT performance configuration is equivalent to this:

Reviewer: ```suggestion
The default settings are equivalent to the following Consul server agent configuration:
```

}
```

When running Consul in production, we recommend configuring the server [performance parameters](/consul/docs/reference/agent/configuration-file/general#performance) with the lowest possible RAFT multiplier setting. This will let Consul servers detect a failed leader and complete leader elections quicker than the default configuration, at the expense of higher resource usage. We recommend the following high-performance configuration for Consul clusters, unless there is a specific reason to use the slower default configuration - for example, running on low-end hardware, on a high-latency network, or on instances sharing workloads.

The recommended high performance configuration looks like this:

> ```json

Reviewer: Shouldn't there by HCL too?
